Feature/ui simplification

n18l opened the pull request:
The overall feedback from initial beta testing was that Search Tailor's UI was too complicated and unintuitive. Given that browser extensions should ideally be usable with very little preamble, I've undertaken a large refactor of the UI based on the feedback received.

There are still some outstanding issues with this rework, but overall it's much more straightforward, both from a UI and data architecture standpoint.

**Major changes as of this writing:**

- "Treatment Groups" have been removed in favour of a simple list of "Tailoring Entries" (much like earlier iterations of the UI)
- Each entry can contain as many domains as desired by use of a "token" field (aka Chips, Tags, etc)
- Each entry can be styled as desired via options in its settings panel
- Styles applied to an entry affect all of the domains listed in the entry
- Styling an entry's opacity to 0 hides it completely within your search results

**Known issues as of this writing:**

- Loading indicator is not visible
- The scroll bar occasionally obscures certain interactions
- The domain field cannot be tabbed out of (this is specific to the TokenField library)

**Planned features as of this writing:**

- Quick actions for opacity setting to jump to 0/100%

**Under consideration as of this writing:**

- Only allowing one settings drawer to be open at a time

Commit:
Update homepage URL and contact email address

Commit:
Update addon icons

Commit:
Update addon description

Commit:
Add patch number to version

Commit:
Fix syncing entries added to new groups

Fixed a bug causing entries moved to newly created groups to have their
treatment information corrupted

Commit:
Merge branch 'bugfix/new-group-entry-syncing' into develop

Commit:
Add tokenfield package

Commit:
Add option to reset to default extension data

Commit:
Temporarily disable the popup interface

Commit:
Refactor treatments around new data structure

Commit:
Move TailoringEntry references to v1 namespace

Commit:
Add token input for domain entries

Commit:
Ensure entry UI order is preserved when saving

Commit:
Remove restrictions on empty tailoring entries

Commit:
Style the token input field for domains

Commit:
Improve event handling for token field

Commit:
Remove entry objects from runtime data on delete

Commit:
Hide popup loading indicator post-load

Commit:
Reenable addition of new entries

Commit:
Use keydown event rather than keypress

Commit:
Streamline token spacing with single variable

Commit:
Implement per-entry settings drawers

Commit:
Remove contents of dist folder on clean task

Updated `clean` task to only remove the contents of the folder
rather than deleting the folder entirely, which caused issues with
open terminal tabs

Commit:
Reenable entry colour customization

Commit:
Enable addition of token field items on blur

Commit:
Fix bug in entry index identification

Commit:
Use ES6 module syntax for imports

Commit:
Import addonFunction methods granularly

Commit:
Move query alias functions to addonFunctions

Commit:
Refactor addonFunction exports

- Resolved import/export-related ESLint errors
- Pruned unused addonFunctions

Commit:
Animate colour picker modal open/close

Commit:
Add UI for entry opacity control

Commit:
Enable entry opacity inputs

Commit:
Add tooltips to entry settings

Commit:
Enable informational tooltips for entry opacity

Commit:
Ensure new Tailoring Entries have unique settings

Fixed a bug that caused all new Tailoring Entries to refer to the
same default settings object, making changes to one entry affect
all entries

Commit:
Include opacity when resetting treatments

Fixed a bug where opacity settings could remain applied when
deleting a tailoring entry

Commit:
Perform per-entry tooltip initialization

Fixed a bug where tooltips would not exist for newly created
tailoring entries until the popup was reopened

Commit:
Reimplement focus effect for domain field

Commit:
Remove dead code references

Commit:
Prune deprecated files

Commit:
Move & document isValidJson function

Commit:
Remove dead code references from popup scripting

Commit:
Refactor colour variables for consistency

Commit:
Update addon icon

Commit:
Update manifest for testing

Commit:
Change icon colour

nbrombal commented:
Entries can't be dragged when the popup is pinned the to the overflow menu. Dragging causes it to close the menu without saving changes.

Commit:
Add default web-ext configuration

Commit:
Restructure search engine setting data

Commit:
Update default entry data for testing

Commit:
Disable entry container scrollbar

Fixed an issue with scrollbar overlap of entry controls by
disabling the scrollbar entirely (what could go wrong?)

Commit:
Refactor primary content script

Commit:
Update opacity settings on input

Commit:
Trigger opacity container tooltip on input element

Commit:
Organize entry settings into groups

Commit:
Enable quick toggling of entry opacity

Commit:
Update border colour icon

Commit:
Update popup window size

Commit:
Only allow one open settings drawer at a time

Commit:
Replace title with aria-label

Resolved an issue with tooltips competing with default title
text on hover

Commit:
Update opacity tooltips

Updated the tooltips for each entry's opacity controls to clarify
their purpose and effects

Commit:
Add function to extract values from a HSLA strings

Commit:
Update default entry

- The default tailoring entry now points to everyone's favourite
online encyclopedia
- The default tailoring entry now uses full-opacity colour values

Commit:
Add styles for disabled entry settings buttons

Commit:
Hide colour picker alpha slider

- Removed the colour picker alpha slider to help prevent
confusion between background/border opacities and overall entry
opacity
- The slider is hidden via CSS rather than disabled with JS to
preserve the ability to apply alpha transparency programmatically

Commit:
Add button for toggling an entry's "colorization"

- Users can now control an entry's "colorization", which toggles
the transparency of an entry's background & border treatment colours
between 1 and 0 hiding their effects without losing colour info

Commit:
Refactor action bar styles and functionality

nbrombal commented:
This overhaul is complete and the popup interface is stable. I'll address remaining known issues with their own tickets and branch as much as possible.
